pkg/provisioner/ironic/configdrive_test.go

```diff
@@ -0,0 +1,134 @@
+package ironic
+
+import (
+	"testing"
+
+	"github.com/gophercloud/gophercloud/openstack/baremetal/v1/nodes"
+	"github.com/gophercloud/gophercloud/openstack/baremetalintrospection/v1/introspection"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	"github.com/metal3-io/baremetal-operator/pkg/bmc"
+	"github.com/metal3-io/baremetal-operator/pkg/provisioner"
+	"github.com/metal3-io/baremetal-operator/pkg/provisioner/fixture"
+	"github.com/metal3-io/baremetal-operator/pkg/provisioner/ironic/clients"
+	"github.com/metal3-io/baremetal-operator/pkg/provisioner/ironic/testserver"
+)
+
+func TestEmpty(t *testing.T) {
+	nodeUUID := "33ce8659-7400-4c68-9535-d10766f07a58"
+
+	cases := []struct {
+		name     string
+		hostData provisioner.HostConfigData
+		expected nodes.ConfigDrive
+	}{
+		{
+			name:     "empty",
+			hostData: fixture.NewHostConfigData("", "", ""),
+			expected: nodes.ConfigDrive{},
```

Comment:
As mentioned below IIUC this should always contain the default metaData, so this isn't expected to be empty, and in the "only meta data" case we should rename the case to "extra meta data" or something, to prove user-provided meta-data is merged in with the defaults.

Reply:
I'd be worried about changing this behavior tbh. Don't we want to leave an option of not having a configdrive at all?